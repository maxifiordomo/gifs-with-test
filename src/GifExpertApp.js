import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp = ({ defaultCategories = [''] }) => {

    const [categories, setCategories] = useState(defaultCategories)

    /*
    const handleAdd = () => {
        setCategories([...categories, 'hunter x hunter']);

        setCategories(cats => [...cats,'Hunter x Hunter]);
    }
    */

    return (
        <>
            <h2>Gif App</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                {
                    categories.map(category => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    )
}
