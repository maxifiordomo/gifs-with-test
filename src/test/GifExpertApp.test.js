//Hacer match con el snapshot
import React from 'react';
import { shallow } from "enzyme"
import { GifExpertApp } from "../GifExpertApp"


describe('test sobre el componente <GifExpertApp/>', () => {
    test('Debe de coincidir con el snapshot', () => {
        const wrapper = shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot();
    })
    test('debe de mostrar una lista de categorias', () => {
        const categories = ['One punch', 'DBZ'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)

        //En este test literalmente estoy dibujando una lista de categorias aleatoria que logra comprobarme si el componente <GifGrid/> recibe correctamente las categorias mandadas
    })


})
