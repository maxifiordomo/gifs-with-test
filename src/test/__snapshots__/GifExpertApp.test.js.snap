// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test sobre el componente <GifExpertApp/> Debe de coincidir con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category=""
      key=""
    />
  </ol>
</Fragment>
`;

exports[`test sobre el componente <GifExpertApp/> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="DBZ"
      key="DBZ"
    />
  </ol>
</Fragment>
`;
